qiskit/providers/ibmq/experiment/experimentservice.py

```diff
@@ -353,14 +363,25 @@ def analysis_results(
                     qual = qual.value  # type: ignore[assignment]
                 qual_str = qual if op == 'eq' else "{}:{}".format(op, qual)
                 quality_list.append(qual_str)
+
+        tags_filter = None
+        if tags:
+            if tags_operator.upper() == 'OR':
+                tags_filter = 'any:' + ','.join(tags)
+            elif tags_operator.upper() == 'AND':
+                tags_filter = 'contains:' + ','.join(tags)
```

Comment:
I don't see anything testing this behavior. The default `tags_operator` value is `OR` and that's going to be exercised in the existing tests, but if you're adding a new feature here there should be test coverage for this case as well, and probably the `ValueError` case below.

Reply:
[Updated!](https://github.com/Qiskit/qiskit-ibmq-provider/pull/1002/files#diff-9275794e4268a2c8dd165487dab5c87c3d271f9692458ac71902381b701725b6R630-R671) Also both experiment and (new) analysis result tests check `ValueError` case.